---
id: 6606b224a69a293f98f8db8f
title: Passo 17
challengeType: 20
dashedName: step-17
---

# --description--

O método `.index()` lança uma exceção `ValueError` quando o valor não é encontrado.Para evitar que o programa pare a execução, você deve aninhar esta linha de código dentro de um bloco `try`.A declaração `try` é usada para encapsular código que pode gerar uma exceção.A cláusula `except`, por outro lado, oferece um código alternativo para executar se uma exceção ocorrer:

```py
try:
    <code>
except:
    <code>
```

Coloque a atribuição de `col` dentro de um bloco `try`.Então, crie uma cláusula `except` e preencha seu corpo com `pass`.

# --hints--

Você deve colocar a atribuição de `col` dentro de um bloco `try` e criar uma cláusula `except`.

```js
({ test: () => assert(runPython(`_Node(_code).find_class("Board").find_function("find_empty_cell").find_for_loops()[0].find_bodies()[0].is_equivalent("try:\\n  col = contents.index(0)\\nexcept:\\n  pass")`)) })
```

# --seed--

## --seed-contents--

```py
class Board:
    def __init__(self, board):
        self.board = board
--fcc-editable-region--
    def find_empty_cell(self):
        for row, contents in enumerate(self.board):
            col = contents.index(0)
--fcc-editable-region--
puzzle = [
  [0, 0, 2, 0, 0, 8, 0, 0, 0],
  [0, 0, 0, 0, 0, 3, 7, 6, 2],
  [4, 3, 0, 0, 0, 0, 8, 0, 0],
  [0, 5, 0, 0, 3, 0, 0, 9, 0],
  [0, 4, 0, 0, 0, 0, 0, 2, 6],
  [0, 0, 0, 4, 6, 7, 0, 0, 0],
  [0, 8, 6, 7, 0, 4, 0, 0, 0],
  [0, 0, 0, 5, 1, 9, 0, 0, 8],
  [1, 7, 0, 0, 0, 6, 0, 0, 5]
]

gameboard = Board(puzzle)
```
